src/diff/children.js

```diff
@@ -183,7 +183,7 @@ export function diffChildren(
 			//
 			// To fix it we make sure to reset the inferred value, so that our own
 			// value check in `diff()` won't be skipped.
-			if (newParentVNode.type == 'option') {
+			if (newParentVNode.type == 'option' && !isHydrating) {
```

Comment:
I think reversing the conditions here so the isHydrating check comes first will reduce the size to ~0.